server/accounts/db.ts

```diff
@@ -1,22 +1,25 @@
 import { db } from 'db';
 import { OxPlayer } from 'player/class';
 import type { OxAccount } from 'types';
+import locales from '../../common/locales';
 
 const addBalance = `UPDATE accounts SET balance = balance + ? WHERE id = ?`;
 const removeBalance = `UPDATE accounts SET balance = balance - ? WHERE id = ?`;
 const safeRemoveBalance = `${removeBalance} AND (balance - ?) >= 0`;
+const addTransaction = `INSERT INTO accounts_transactions (fromId, toId, amount, message) VALUES (?, ?, ?, ?)`;
 
-export async function UpdateBalance(id: number, amount: number, action: 'add' | 'remove', overdraw?: boolean) {
+export async function UpdateBalance(id: number, amount: number, action: 'add' | 'remove', overdraw: boolean, message?: string) {
   return (
     (await db.update(action === 'add' ? addBalance : overdraw ? removeBalance : safeRemoveBalance, [
       amount,
       id,
       amount,
-    ])) === 1
+    ])) === 1 &&
+    (await db.insert(addTransaction, [action === 'add' ? null : id, action === 'add' ? id : null, amount, message ?? locales("no_message")])) === 1
```

Comment:
You can assign the default value of "message" inside the function constructor. However, I would just leave the message blank (null) rather than setting a default.